### PR TITLE
NASA jumpsuit now provides space protection. 

### LINKED ISSUE #38955: The NASA jumpsuit doesn't stop pressure damage in space
Title, the NASA jumpsuit "/obj/item/clothing/under/space" is no longer space worthy since the user will still take pressure damage when worn in space.

To reproduce wear the NASA jumpsuit, shoes, gloves, and a space worthy helmet and jump into space, then watch as you die from the vacuum of space anyway.

Everything about the jumpsuit says it's space worthy, and it used to be so until breaking at some point.

Possibly caused by #37486 but don't trust me on this.

### PR DESCRIPTION
**Pull request by Irafas**
-uniforms are now checked to see if they provide any protection
-NASA jumpsuit now actually provides space protection

:cl:
fix: NASA jumpsuit provides space protection
/:cl:

fixes: #38955 

The jumpsuit doesn't have a unique appearance and just looks like a standard black jumpsuit. I don't really know how to sprite but having it coloured orange with the logos like an actual NASA jumpsuit would be cool.

**_orangetide_**
orange-astronaut-jumpsuit-costume

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -44,11 +44,16 @@
 
 
 /mob/living/carbon/human/calculate_affecting_pressure(pressure)
-	if (wear_suit && head && istype(wear_suit, /obj/item/clothing) && istype(head, /obj/item/clothing))
```
Comment (Jalleo) [inline code/modules/mob/living/carbon/human/life.dm]: couldn't you just do a 
var/check_this_one = null
if ((wear_suit &| w_uniform) && head) INCLUDE THE ISTYPING HERE
     if(wear_suit.clothing_flags == STOPPRESSUREDAMAGE)
            check_this_one = wear_suit
      else if(same as above but for w_uniform)
      else return pressure
then do the helmet check? AKA slightly less on the processing

**Comment by optimumtact** [comment]
This adds a lot of code just to support a single jumpsuit into being space worthy.

Just make the jumpsuit into a spacesuit if it bothers you that much

:-1:

**Comment by Irafas** [comment]
Yeah that's what I was worried about. There's a lot of added code just to add one function to a suit nobody wears anyway. I used the profiler to look at if it made an impact but that didn't help much as it showed no difference, and it was only me on the server as well so it wasn't a very good sample size.

There's already a NASA spacesuit, though its dark red, so no much point making this one a suit. I'm just going to close this PR as it fixes an issue that doesn't have any impact and does so at the expense of adding more code to a frequently called proc.

**Comment by Irafas** [comment]
That's true. I had forgotten about pressurization potions.

**Pull request closed without merging**